lib/rom/setup/auto_registration.rb

```diff
@@ -44,21 +44,67 @@ def mappers
     def load_entities(entity)
       Dir[globs[entity]].map do |file|
         require file
-        Inflector.constantize(
-          const_name(component_dirs.fetch(entity), file)
+        klass_name = case
+        when namespace.class == String
+          CustomNamespaceStrategy.new(namespace: namespace, file: file).call
+        when namespace == true
+          WithNamespaceStrategy.new(file: file, directory: directory).call
+        when namespace == false
+          NoNamespaceStrategy.new(file: file, directory: directory, entity: component_dirs.fetch(entity)).call
+        end
```

Comment:
This could simplified like that:

``` ruby
    case namespace
      when String
          CustomNamespaceStrategy.new(namespace: namespace, file: file).call
        when TrueClass
          WithNamespaceStrategy.new(file: file, directory: directory).call
        when FalseClass
          NoNamespaceStrategy.new(file: file, directory: directory, entity: component_dirs.fetch(entity)).call
        end
````